launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -190,6 +190,10 @@
       firstNonEmptyValue(SparkLauncher.DRIVER_EXTRA_CLASSPATH, conf, props) : null;
 
     List<String> cmd = buildJavaCommand(extraClassPath);
+    // Take Thrift Server as daemon
+    if (isThriftServer(mainClass)) {
```

Comment:
Yeah. I think so. If we take a look at `start-thriftserver.sh` then will find in `spark-daemon.sh` we use spark-submit to submit the `org.apache.spark.sql.hive.thriftserver.HiveThriftServer2`. 

Then it comes into spark-class, and spark-class use `org.apache.spark.launcher.Main` to resolve args in which it has:
>if (className.equals("org.apache.spark.deploy.SparkSubmit")) {
        builder = new SparkSubmitCommandBuilder(args);
      } else {
        builder = new SparkClassCommandBuilder(className, args);
      }

So I think we start Thrift Server by spark-submit but at the meantime Thrift Server is sort of daemon process and it should take daemon related options too.